### DISCUSSION TITLE
Stellar Asset Contract Issuer semantics

### DISCUSSION THREAD

**Discussion by sisuresh** [stellar/stellar-protocol#1598] [Core Advancement Proposals]
The Stellar Asset Contract special cases the issuer because issuers can't have a trustline to their own asset, which leads to two related details that should be revisited - 

1. If the issuer is the sender in a `transfer`, the `amount` will be minted. If the issuer is the receiver in a `transfer`, the `amount` will be burned. This follows classic Stellar semantics.
2. In the two above cases where the token is minted or burned using `transfer`, we emit the `transfer` event (instead of `mint` or `burn`).

These two issues lead to additional difficulties when it comes to tracking token balances. We have two actions we can take here - 

1. Stop special casing the issuer in the SAC and fail when the issuer is the sender or receiver.
2. Continue allowing the issuer case, but emit `mint` or `burn` instead of `transfer` when the issuer is involved.

**Comment by leighmcculloch** [comment]
This has been rolled into CAP-67 captured here:
- https://github.com/stellar/stellar-protocol/discussions/1553
